HttplugFactory.php

```diff
@@ -95,11 +93,14 @@ protected function evaluateCondition($condition)
             $evaluatedCondition = true;
 
             // Immediately stop execution if the condition is false
-            for ($i = 0; $i < count($condition) && false !== $evaluatedCondition; ++$i) {
-                $evaluatedCondition &= $this->evaluateCondition($condition[$i]);
```

Comment:
I will only "look nice". You will still do the same number of comparisons. The current solution does not support assoc. arrays. 

Anyhow. I updated the PR so it gets the best of both worlds... What do you think?